pages/blog/graphql-authz.mdx

```diff
@@ -0,0 +1,236 @@
+
+The benefits of the wrapping approach are described in previous paragraphs, however, there are also some drawbacks. For example, with GraphQL AuthZ post-execution rules there is no ability to fail the request early because post-execution rules are executed after all resolvers are executed. GraphQL Shield, on the other hand, can fail the request during the execution phase which happens before the post-execution phase but later than the pre-execution phase. On the other hand, post-execution rules have the benefit of accessing the resolved value of the field they are specified on and, furthermore, even the sibling fields that are specified via the rules selection set.
+
+The middleware approach of GraphQL shield is missing these possibilities because authorization logic is executed in the context of field resolvers and has access only to the parent object which is the value returned from the parent field resolver. At that stage, the wrapped field resolvers have not been executed yet.
+
+Another feature in GraphQL Shield is the built-in contextual caching mechanism for rules. At the moment, GraphQL AuthZ has no built-in caching, but you can implement it yourself within the GraphQL AuthZ rules, or in the future caching could even become a built-in feature. (pull requests are more than welcome)
+
+## Which approach to choose?
+
+To sum up use-cases and different authorization approaches, here is the tip:
+
+- If you have very few places that should be covered by authorization and you don't plan to increase such places, you can just add authorization logic right inside resolvers to not overcomplicate things.
+- If all of your authorization logic doesn't depend on the data and shouldn't perform complex calculations, you can use [operation-field-permissions](https://www.envelop.dev/plugins/use-operation-field-permissions) [Envelop](https://www.envelop.dev/) plugin.
+- If your authorization logic contains some calculations which are the same for many fields of your schema, you can use [GraphQL Shield](https://github.com/maticzav/graphql-shield) to leverage the built-in contextual caching mechanism.
+- If your authorization logic heavily depends on the data or you want to use schema directives to attach auth rules, you can use [GraphQL-AuthZ](https://github.com/AstrumU/graphql-authz) to leverage the 'GraphQL schema as a data source' pattern and post-execution rules.
+
+## Conclusion
+
+GraphQL AuthZ is a new approach to doing GraphQL native authorization. We are happy that we can finally share this library with the community and keen to learn about the ways it might be used within your next project! Don't hesitate to contact us for questions or to evaluate whether this library might be a fit for your architecture.
```

Comment:
```suggestion
GraphQL AuthZ is a new approach for applying GraphQL native authorization. We are happy that we can finally share this library with the community and keen to learn about the ways it might be used within your next project! Don't hesitate to contact us for questions or to evaluate whether this library might be a fit for your architecture.
```